pkg/extensions/tbtc/tbtc.go

```diff
@@ -97,6 +134,270 @@ func (t *tbtc) monitorRetrievePubKey(
 	return nil
 }
 
+func (t *tbtc) monitorProvideRedemptionSignature(
+	ctx context.Context,
+	actBackoffFn backoffFn,
+	timeout time.Duration,
+) error {
+	monitoringStartFn := func(
+		handler depositEventHandler,
+	) (subscription.EventSubscription, error) {
+		// Start right after a redemption has been requested or the redemption
+		// fee has been increased.
+		return t.chain.OnDepositRedemptionRequested(handler)
+	}
+
+	monitoringStopFn := func(
+		handler depositEventHandler,
+	) (subscription.EventSubscription, error) {
+		// Stop in case the redemption signature has been provided by someone else.
+		signatureSubscription, err := t.chain.OnDepositGotRedemptionSignature(handler)
+		if err != nil {
+			return nil, err
+		}
+
+		// Stop in case the redemption proof has been provided by someone else.
+		redeemedSubscription, err := t.chain.OnDepositRedeemed(handler)
+		if err != nil {
+			return nil, err
+		}
+
```

Comment:
It is tricky but shouldn't we also close the current monitoring goroutine in case of `RedemptionRequested` event? It means that someone fee-bumped and we'll have another instance of this goroutine created.

Alternatively - and I am not sure if it is not a better solution - we shouldn't create another monitoring goroutine for `provideRedemptionSignature` in case there is already one working. With this approach we can address the scenario described above but also event duplicates.

Reply:
This goroutine works only when a deposit is in the `AWAITING_WITHFRAWAL_SIGNATURE` state. Once the signature is submitted this goroutine stops and the deposit switches to `AWAITING_WITHDRAWAL_PROOF` state. Fee bumps are only possible in the latter state.